### PR TITLE
[Communication] - ALL - Add subscription configuration to tests.yml

### PR DESCRIPTION
**Pull request by minnieliu**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/communication/tests.yml [hunk 1]
```diff
@@ -16,16 +16,16 @@ extends:
   template: ../../eng/pipelines/templates/stages/archetype-sdk-tests.yml
   parameters:
     ServiceDirectory: communication
-    Clouds: ACS_Public
     CloudConfig:
-      ACS_Public:
-        SubscriptionConfiguration: $(sub-config-communication-services-cloud-test-resources)
+    Public:
```
Comment (beltr0n) [inline sdk/communication/tests.yml:20]: does this need to be indented?

Comment (sacheu) [reply sdk/communication/tests.yml:20]: Thanks Ben for catching this.

File: sdk/communication/tests.yml [hunk 2]
```diff
@@ -16,16 +16,16 @@ extends:
...
+      SubscriptionConfigurations:
+        - $(sub-config-azure-cloud-test-resources)
+        - $(sub-config-communication-services-cloud-test-resources-test)
+        - $(sub-config-communication-services-cloud-test-resources-net)
+    Int:
```
Comment (sacheu) [inline sdk/communication/tests.yml:25]: Remove the Int setting here.

File: sdk/communication/tests.yml [hunk 3]
```diff
@@ -16,16 +16,16 @@ extends:
...
+      SubscriptionConfiguration: $(sub-config-communication-int-test-resources)
+    Clouds: Public,Int
```
Comment (sacheu) [inline sdk/communication/tests.yml]: Remove the Int

File: sdk/communication/test-resources.json [hunk 4]
```diff
@@ -46,13 +72,29 @@
...
+        "type": "string",
+        "value": "[concat('https://', parameters('baseName'), '-', parameters('endpointPrefix'), parameters('communicationServicesEndpointSuffix'))]"
+      },
         "RESOURCE_GROUP_NAME": {
             "type": "string",
             "value": "[resourceGroup().Name]"
         }
     }
-}
+}
```
Comment (sacheu) [inline sdk/communication/test-resources.json:100]: nit: add extra line

**Review by benbp — APPROVED** [review]

File: sdk/communication/test-resources.json [hunk 5]
```diff
@@ -13,11 +13,37 @@
...
+            "type": "string"
+        },
         "testApplicationOid": {
             "type": "string",
             "metadata": {
                 "description": "The client OID to grant access to test resources."
             }
+        },
+        "tenantId": {
+            "type": "String",
```
Comment (beltr0n) [inline sdk/communication/test-resources.json:31]: String should be lowercase, there's a couple spots where this needs to be done

**Review by beltr0n — CHANGES_REQUESTED** [review]

File: sdk/communication/test-resources.json [hunk 6]
```diff
@@ -46,10 +72,26 @@
...
+        },
+        "AZURE_CLIENT_SECRET": {
+            "type": "string",
+            "value": "[parameters('testApplicationSecret')]"
+        },
         "COMMUNICATION_CONNECTION_STRING": {
             "type": "string",
             "value": "[listKeys(resourceId('Microsoft.Communication/CommunicationServices',variables('uniqueSubDomainName')), '2020-08-20-preview').primaryConnectionString]"
         },
+        "COMMUNICATION_SERVICE_ENDPOINT": {
```
Comment (sacheu) [inline sdk/communication/test-resources.json:91]: this is not needed in .net

File: sdk/communication/test-resources.json [hunk 7]
```diff
@@ -13,11 +13,37 @@
             "defaultValue": "communication",
             "type": "string"
         },
+        "communicationServicesEndpointSuffix": {
```
Comment (sacheu) [inline sdk/communication/test-resources.json:16]: this is not needed in .net.

File: sdk/communication/tests.yml [hunk 8]
```diff
@@ -16,16 +16,14 @@ extends:
...
   parameters:
     ServiceDirectory: communication
-    Clouds: ACS_Public
     CloudConfig:
-      ACS_Public:
-        SubscriptionConfiguration: $(sub-config-communication-services-cloud-test-resources)
+      Public:
+        SubscriptionConfigurations:
+          - $(sub-config-azure-cloud-test-resources)
+          - $(sub-config-communication-services-cloud-test-resources-test)
```
Comment (sacheu) [inline sdk/communication/tests.yml:23]: i suggest rename this to something like `sub-config-communication-services-cloud-test-resources-common`

File: sdk/communication/test-resources.json (hunk 5 above)
Comment (minnieliu) [reply sdk/communication/test-resources.json]: Done!

File: sdk/communication/tests.yml (hunk 8 above)
Comment (minnieliu) [reply sdk/communication/tests.yml:23]: We need to change the keyvault and the variable name first right?

**Review by sacheu — APPROVED** [review]
Thanks Minnie. Look good.

**Review by beltr0n — APPROVED** [review]

**Pull request merged**